### PR TITLE
ITSASU-805: Create Agent Overseas property CYA

### PR DESCRIPTION
**Pull request by sb-dev**

### CODE REVIEW

File: it/controllers/agent/business/OverseasPropertyCheckYourAnswersControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    "return NOT_FOUND" when {
+      "the save and retrieve feature switch is disabled" in {
+        Given("I setup the Wiremock stubs")
+        AuthStub.stubAuthSuccess()
+        IncomeTaxSubscriptionConnectorStub.stubSubscriptionData(subscriptionData())
+        IncomeTaxSubscriptionConnectorStub.stubGetSubscriptionDetails(OverseasProperty, NO_CONTENT)
+        And("save & retrieve feature switch is enabled")
+        enable(SaveAndRetrieve)
```
[the remark this answers is not in the document]
Comment (aug24) [reply it/controllers/agent/business/OverseasPropertyCheckYourAnswersControllerISpec.scala:63]: Check out the `withFeatureSwitch` utility I added?  Makes everything clearer, and always turns off the specified feature switch after the test.

[the remark this answers is not in the document]
Comment (scullythornton1) [reply it/controllers/agent/business/OverseasPropertyCheckYourAnswersControllerISpec.scala:63]: > Check out the `withFeatureSwitch` utility I added? Makes everything clearer, and always turns off the specified feature switch after the test.

Hi, Justin, where can I find that utility, cannot seem to see it in Sign up FE repo?

File: it/controllers/agent/business/OverseasPropertyCheckYourAnswersControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        res should have(
+          httpStatus(SEE_OTHER),
+          redirectURI(taskListURI)
+        )
+
+        IncomeTaxSubscriptionConnectorStub.verifySaveOverseasProperty(OverseasPropertyModel(accountingMethod = Some(Cash), confirmed = true), Some(1))
+      }
+    }
+
+    "return INTERNAL_SERVER_ERROR" when {
```
[the remark this answers is not in the document]
Comment (sb-dev) [reply it/controllers/agent/business/OverseasPropertyCheckYourAnswersControllerISpec.scala:98]: The error comes from the `withOverseasProperty` function. It get thrown when the overseas property cannot get retrieved.

File: it/controllers/agent/business/OverseasPropertyCheckYourAnswersControllerISpec.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (sb-dev) [reply it/controllers/agent/business/OverseasPropertyCheckYourAnswersControllerISpec.scala:63]: I can add the utility.

**Pull request merged**